### PR TITLE
[Designer/templating] Fixes + add toString

### PR DESCRIPTION
**Pull request by dclaux**
## Related Issue
Fixes https://github.com/MicrosoftDocs/AdaptiveCards/issues/251

## Description
- Actually create a copy of the expression nodes which were previously overwritten causing the same data object to be reused for all items in an array
- Introduce toString to support basic explicit conversion of numbers into string

## How Verified
Verified in:
- adaptivecards-designer-app
- Web site running locally

## How to test
Card payload:
```json
{
    "$schema": "http://adaptivecards.io/schemas/adaptive-card.json",
    "type": "AdaptiveCard",
    "version": "1.0",
    "body": [
        {
            "type": "Container",
            "items": [
                {
                    "type": "ColumnSet",
                    "columns": [
                        {
                            "type": "Column",
                            "width": "auto",
                            "items": [
                                {
                                    "type": "TextBlock",
                                    "text": "{if(changePriceRatio >= 0, '▲', '▼')}",
                                    "color": "{if(changePriceRatio >= 0, 'good', 'attention')}"
                                }
                            ]
                        },
                        {
                            "type": "Column",
                            "width": "stretch",
                            "items": [
                                {
                                    "type": "TextBlock",
                                    "text": "{name}"
                                }
                            ]
                        },
                        {
                            "type": "Column",
                            "width": "stretch",
                            "items": [
                                {
                                    "type": "TextBlock",
                                    "text": "{toString(lastPrice)}",
                                    "horizontalAlignment": "Center"
                                }
                            ],
                            "horizontalAlignment": "Center"
                        },
                        {
                            "type": "Column",
                            "width": "auto",
                            "items": [
                                {
                                    "type": "TextBlock",
                                    "text": "{changePriceRatio}%",
                                    "color": "{if(changePriceRatio >= 0, 'good', 'attention')}"
                                }
                            ],
                            "horizontalAlignment": "Right"
                        }
                    ]
                }
            ],
            "$data": "{$root}"
        }
    ]
}
```

Data payload:
```json
[
    {
        "name": "Object-1",
        "lastPrice": 1.10762,
        "changePriceRatio": -0.17
    },
    {
        "name": "Object-2",
        "lastPrice": 1578.205,
        "changePriceRatio": -0.68
    },
    {
        "name": "Object-3",
        "lastPrice": 51.475,
        "changePriceRatio": -0.23
    },
    {
        "name": "Object-4",
        "lastPrice": 28324,
        "changePriceRatio": 0.35
    },
    {
        "name": "Object-5",
        "lastPrice": 9338.87,
        "changePriceRatio": -1.04
    }
]
```

Card should render as follows in the designer in "Preview mode":
image

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/AdaptiveCards/pull/3816)

### CODE REVIEW

File: source/nodejs/adaptivecards-designer-app/src/app.ts [hunk 1]
```diff
@@ -12,13 +12,11 @@ import "./app.css";
 
 window.onload = function() {
     // Uncomment to enabled preview features such as data binding
-    /*
```
Comment (shalinijoshi19) [inline source/nodejs/adaptivecards-designer-app/src/app.ts]: Are we uncommenting this to expose the databinding feature by default now?

**Comment by shalinijoshi19** [comment]
@dclaux, @matthidinger, guys, question: how/when are we expecting to spin up a package update for this and also should we update the version up on vnext.adaptivecards.io for bug bash purposes ?  Also David, can you actually include the sample update that we can use both for the bash as well as to update our documentation site if needed?

File: source/nodejs/adaptivecards-designer-app/src/app.ts (hunk 1 above)
Comment (dclaux) [reply source/nodejs/adaptivecards-designer-app/src/app.ts]: Yes. Also keep in mind that adaptivecards-designer-app is basically a test app.

**Comment by matthidinger** [comment]
@shalinijoshi19 I'd love to try and get an update out tomorrow if the fix is confirmed working in the site. If not we might need a few more days into next week, but we should definitely sync on the overall status of the latest site deployments and close those ends. The Project board has the 4 items I'm hoping to close for the next publish

**Pull request merged**